packages/sanity/src/structure/panes/documentList/PaneContainer.tsx

```diff
@@ -112,7 +103,7 @@ export const PaneContainer = memo(function PaneContainer(
 
   const isSheetListLayout = layout === 'sheetList'
   const paneLayout = isSheetListLayout ? (
-    <DocumentSheetListPane {...props} key={props.pane.id} />
```

Comment:
I think we need to restore this `key` in here, it was added to reset all the table states and effects when switching from one document type to another, forcing the component to re mount.

Reply:
Ah that's fair - I didn't realise! I'll reinstate

- [ ] Add `key` back to `DocumentSheetListPane`